framework/include/functions/PiecewiseConstantFromCSV.h

```diff
@@ -0,0 +1,55 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Function.h"
+#include "PropertyReadFile.h"
+
+/**
+ * Function which provides a piecewise constant field from a CSV file
+ */
+class PiecewiseConstantFromCSV : public Function
+{
+public:
+  static InputParameters validParams();
+
+  PiecewiseConstantFromCSV(const InputParameters & parameters);
+
+  /**
+   * Get the value of the function (based on time only)
+   * \param t The time
+   * \param pt The point in space (x,y,z) (unused)
```

Comment:
why is the point in space "(unused)"?  Don't you need that for option 3 above?

Reply:
oh yeah time is unused